src/sources/prepared_source.rs

```diff
@@ -0,0 +1,217 @@
+use std::sync::Arc;
```

Comment:
a PreparedSource is to be used in cases where we need a TableProvider but the given DataSource doesn't support it. in those cases the PreparedSource will convert that DataSource into the arrow file format on disk, which we can then use to give us a TableProvider.

Reply:
TODO: allow controlling where this is created, many systems have limited space in /tmp